### DISCUSSION TITLE
How to suppress certain warnings?

### DISCUSSION THREAD

**Discussion by bronger** [emacs-lsp/lsp-mode#3835] [Q&A]
When editing a Go file, I see three warnings in the modeline (it says `FlyC:0|3` there).  The affected locations in the code are highlighted. However, the respective “error” (unkeyed fields) is made deliberately by me and I don’t want to change that. Can this specific warning be suppressed on any level (lsp-mode or gopls)?

**Comment by alexander-naumov** [comment]
Yes, sure, you can suppress specific warnings in lsp-mode and gopls to avoid seeing the "unkeyed fields" warning in your modeline. To achieve this, you can customize the `lsp-diagnostic-filter` variable in your Emacs configuration.

The `lsp-diagnostic-filter` variable allows you to customize which types of diagnostics (warnings, errors, hints, etc.) are displayed in the modeline and the buffer. By setting this variable appropriately, you can choose to hide specific diagnostics, including the "unkeyed fields" warning in your case.

**Reply by bronger** [reply]
Thank you, I will give that a try!

In the meantime, I was successful for my specific use case with
```
(lsp-register-custom-settings
 '(("gopls.analyses.composites" nil t)))
```
